borg/key.py

```diff
@@ -397,24 +397,32 @@ class KeyfileKey(KeyfileKeyBase):
     FILE_ID = 'BORG_KEY'
 
     def find_key(self):
-        id = hexlify(self.repository.id).decode('ascii')
-        keys_dir = get_keys_dir()
-        for name in os.listdir(keys_dir):
-            filename = os.path.join(keys_dir, name)
-            with open(filename, 'r') as fd:
-                line = fd.readline().strip()
-                if line.startswith(self.FILE_ID) and line[len(self.FILE_ID) + 1:] == id:
-                    return filename
-        raise KeyfileNotFoundError(self.repository._location.canonical_path(), get_keys_dir())
+        keyfile = os.environ.get('BORG_KEY_FILENAME')
+        if keyfile:
+            return keyfile
+        else:
+            id = hexlify(self.repository.id).decode('ascii')
+            keys_dir = get_keys_dir()
+            for name in os.listdir(keys_dir):
+                filename = os.path.join(keys_dir, name)
+                with open(filename, 'r') as fd:
+                    line = fd.readline().strip()
+                    if line.startswith(self.FILE_ID) and line[len(self.FILE_ID) + 1:] == id:
+                        return filename
+            raise KeyfileNotFoundError(self.repository._location.canonical_path(), get_keys_dir())
```

Comment:
woops, I've already added a KeyfileInvalidError
(and KeyfileIncorrectError does not seem to exist :-))